File: app/lib/src/main/java/com/example/lib/callbacks/CallbackDemo.java
package com.example.lib.callbacks;


public class CallbackDemo {

    int rekhaAdd(int a, int b){
        return a+b;
    }

    //radio h/w data internet--raise interrupt
    void muraliAdd(int a, int b,MyPhNumber callbackPhno){
        try {
            Thread.sleep(5000);//unsure --suspended
            callbackPhno.onSuccess(a+b);  //resumed
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }

    }

}

File: app/src/main/java/com/example/composebosch/kotlinexs/callbacks/CallbackDemo.kt
package com.example.composebosch.kotlinexs.callbacks

import kotlinx.coroutines.delay

class CallbackDemo {

    fun rekaAdd(a:Int,b:Int):Int{
        return a+b
    }
    fun muraliAdd(a:Int,b:Int,myPhno:MyPhoneNum){
        Thread.sleep(5000)
        myPhno.onSuccess(a+b)
    }
}